### PR TITLE
feat: set up production canister

### PR DESCRIPTION
**Pull request by rvanasa**
Adds a production canister deployed on the fiduciary subnet with DFINITY-managed API keys. 

In addition, this PR does a significant refactor of the Motoko E2E tests to allow testing backwards compatibility on mainnet. I also included some CI changes such as running the example dfx commands as part of the `e2e` workflow.

### CODE REVIEW

File: scripts/e2e [hunk 1]
```diff
@@ -2,11 +2,13 @@
...
-    dfx deploy evm_rpc_fiduciary --argument "(record {nodesInSubnet = 28})" --mode reinstall -y &&
-    dfx deploy e2e_rust --mode reinstall -y &&
+    dfx deploy evm_rpc --argument "(record {nodesInSubnet = 28})" --mode reinstall -y &&
+    dfx deploy evm_rpc_staging_13_node --argument "(record {nodesInSubnet = 13})" --mode reinstall -y &&
+    dfx deploy evm_rpc_staging_fiduciary --argument "(record {nodesInSubnet = 28})" --mode reinstall -y &&
+    npm run generate &&
+    dfx deploy e2e_rust &&
     dfx canister call e2e_rust test &&
-    dfx deploy e2e_motoko --mode reinstall -y &&
+    dfx deploy e2e_motoko &&
```
Comment (rvanasa) [inline scripts/e2e:11]: I came across what seems like a heisenbug or corner case with `dfx deploy` where using `--mode reinstall` on the first deployment causes a mysterious `ic-cdk-bindgen` error message. This occasionally happened on CI and other times on my local machine, but it was unclear what was specifically causing this. Just wanted to quickly mention this somewhere, since it took some trial and error to narrow down the issue.

Here's the error message for posterity:

```
thread 'main' panicked at 'Cannot find candid path: NotPresent'
```

The solution was to remove the `--mode reinstall` flag from `dfx deploy`.

File: e2e/motoko/Main.mo [hunk 2]
```diff
@@ -10,21 +11,46 @@ import Buffer "mo:base/Buffer";
 import Evm "mo:evm";
 
 shared ({ caller = installer }) actor class Main() {
+
+    type TestCategory = { #staging; #production };
+
+    type SubnetTarget = (Nat, Nat);
+    // (`subnet name`, `canister module`, `canister type`, `subnet`)
```
Comment (THLO) [inline e2e/motoko/Main.mo:18]: I'm not sure what this comment refers to.

File: e2e/motoko/Main.mo [hunk 3]
```diff
@@ -10,21 +11,46 @@ import Buffer "mo:base/Buffer";
...
+    let defaultSubnet = ("13-node", 13, 99_330_400);
+    let fiduciarySubnet = ("fiduciary", 28, 239_142_400);
+
+    // (`nodes in subnet`, `expected cycles for JSON-RPC call`)
```
Comment (THLO) [inline e2e/motoko/Main.mo:22]: Same here.

File: scripts/e2e (hunk 1 above)
Comment (THLO) [reply scripts/e2e:11]: Heisenbug.   :smile: 

Is this something we should bring up with the SDK team?

**Review by THLO — APPROVED** [review]

File: e2e/motoko/Main.mo (hunk 3 above)
Comment (rvanasa) [reply e2e/motoko/Main.mo:22]: They were swapped with each other; fixed.
